cmake/add_dcap_client_target.cmake

```diff
@@ -0,0 +1,115 @@
+    endif ()
+
+    if (CMAKE_BUILD_TYPE MATCHES Debug OR CMAKE_BUILD_TYPE MATCHES RelWithDebugInfo)
+        set(IMPORT_BUILD_TYPE Debug)
+    elseif (CMAKE_BUILD_TYPE MATCHES Release OR CMAKE_BUILD_TYPE MATCHES MinSizeRel)
+        set(IMPORT_BUILD_TYPE Release)
+    else ()
+        message(FATAL_ERROR "Unsupported CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}, import_dcap_client needs to be updated to support it.")
+    endif ()
+
+    # Initialize the null list of dependencies for the target
+    set(DEPENDENCIES "")
+
+    # Define the DCAP provider paths
+    set(AZURE_DCAP_QUOTEPROV ${CMAKE_SOURCE_DIR}/prereqs/nuget/Microsoft.Azure.DCAP.Client/lib/${IMPORT_BUILD_TYPE}/dcap_quoteprov.dll)
+    set(AZURE_DCAP_QUOTEPROV_SYMBOLS ${CMAKE_SOURCE_DIR}/prereqs/nuget/Microsoft.Azure.DCAP.Client/lib/${IMPORT_BUILD_TYPE}/dcap_quoteprov.pdb)
+
+    # Note that for all of dcap_quoteprov's own dependencies, we always take the Release build.
+    # The Debug versions don't come with symbols anyway and take a dependency on the debug msvcr110d.dll
+    # which is not provided by the vcredist_x64 package and will cause runtime load faiures.
+    set(LIBCURL ${CMAKE_SOURCE_DIR}/prereqs/nuget/curl.redist/build/native/bin/v110/x64/Release/dynamic/libcurl.dll)
+    set(LIBSSH2 ${CMAKE_SOURCE_DIR}/prereqs/nuget/libssh2.redist/build/native/bin/v110/x64/Release/dynamic/cdecl/libssh2.dll)
+    set(LIBEAY32 ${CMAKE_SOURCE_DIR}/prereqs/nuget/openssl.redist/build/native/bin/v110/x64/Release/dynamic/cdecl/libeay32.dll)
+    set(SSLEAY32 ${CMAKE_SOURCE_DIR}/prereqs/nuget/openssl.redist/build/native/bin/v110/x64/Release/dynamic/cdecl/ssleay32.dll)
+    set(ZLIB ${CMAKE_SOURCE_DIR}/prereqs/nuget/zlib.redist/build/native/bin/v110/x64/Release/dynamic/cdecl/zlib.dll)
+
+    # No-op if the DCAP provider is not found
+    if (NOT EXISTS ${AZURE_DCAP_QUOTEPROV})
+        message (WARNING "dcap_quoteprov dependencies were not found, may not execute successfully.")
+    else ()
+        if (NOT EXISTS ${LIBCURL})
+            message(FATAL_ERROR "Found Azure dcap_quoteprov, but could not find its dependency ${LIBCURL}, aborting.")
+        endif ()
+        if (NOT EXISTS ${LIBSSH2})
+            message(FATAL_ERROR "Found Azure dcap_quoteprov, but could not find its dependency ${LIBSSH2}, aborting.")
+        endif ()
+        if (NOT EXISTS ${LIBEAY32})
+            message(FATAL_ERROR "Found Azure dcap_quoteprov, but could not find its dependency ${LIBEAY32}, aborting.")
+        endif ()
+        if (NOT EXISTS ${SSLEAY32})
+            message(FATAL_ERROR "Found Azure dcap_quoteprov, but could not find its dependency ${SSLEAY32}, aborting.")
+        endif ()
+        if (NOT EXISTS ${ZLIB})
+            message(FATAL_ERROR "Found Azure dcap_quoteprov, but could not find its dependency ${ZLIB}, aborting.")
+        endif ()
+
+        # Add copy actions for each of the dependencies
+        add_custom_command(
```

Comment:
Instead of doing this we should be calling the install script provided by the Azure DCAP client nuget package.